R/ssm_to_proteinpaint.R

```diff
@@ -0,0 +1,141 @@
+                refseq = RefSeq, aachange = HGVSp_Short, origin = Mutation_Status,
+                disease = pathology, mutant_in_tumor = t_alt_count, 
+                total_in_tumor = t_depth, mutant_in_normal = n_alt_count, 
+                total_in_normal = n_depth, patient = patient_id, 
+                sample = Tumor_Sample_Barcode, sampletype = time_point,
+                REF = Reference_Allele) %>% 
+    mutate(class = recode(class,
+                          Missense_Mutation = "missense",
+                          In_Frame_Del = "proteinDel",
+                          In_Frame_Ins = "proteinIns",
+                          Nonsense_Mutation = "nonsense",
+                          Nonstop_Mutation = "nonsense",
+                          Silent = "silent",
+                          Splice_Region = "splice_region",
+                          Splice_Site = "splice",
+                          Frame_Shift_Del = "frameshift",
+                          Frame_Shift_Ins = "frameshift"))
+  
+  # remove version names in refseq column
+  maf$refseq = maf$refseq %>% 
+    gsub("\\.[0-9]+,", ",", .) %>% 
+    gsub("\\.[0-9]+$", "", .)
+  
+  # if alt allele is heterogeneous alternative, show both allele separated 
+  # by comma (like in VCF files)
+  is_het_alt = ! maf$REF == maf$Tumor_Seq_Allele1
+  maf$ALT = is_het_alt %>% 
+    { paste( maf$Tumor_Seq_Allele1[.], maf$Tumor_Seq_Allele2[.], sep = "," ) } %>% 
+    replace(maf$Tumor_Seq_Allele2, is_het_alt, .)
+  
+  # add VAF (variant allele fraction)
+  maf = mutate(maf, VAF = mutant_in_tumor/total_in_tumor)
+  
+  # keep only columns that are meaningful for proteinpaint
+  maf = dplyr::select(maf, gene, refseq, chromosome, start, aachange, class, disease, origin,
+                      patient, sample, sampletype, mutant_in_tumor, total_in_tumor, 
+                      mutant_in_normal, total_in_normal, REF, ALT, VAF)
+  
+  # filter out rows that don't contain required column values and output warning message
+  required_cols = dplyr::select(maf, gene, refseq, chromosome, start, aachange, class) %>% 
+    mutate(i = row_number()) %>% 
+    tidyr::drop_na()
+  required_cols = apply(required_cols != "", 1, all) %>% 
+    dplyr::filter(required_cols, .)
+  if(print_removed_rows){  # if only removed rows should be returned (for checking purpose)
+    removed_rows = 1:nrow(maf) %>% 
+      "["(! . %in% required_cols$i) %>% 
+      slice(maf, .)
+    return(removed_rows)
+  }
+  num_removed_rows = (nrow(maf) - nrow(required_cols)) %>% 
+    format(big.mark="'")
+  k = format( nrow(maf), big.mark="'" ) %>% 
+    gettextf("Warning: %s rows out of %s were removed from the output table because there were missing values in required columns. Run `ssm_to_proteinpaint` again with `print_removed_rows = TRUE` to see these rows.",
+             num_removed_rows, .)
+  message(k)
+  maf = slice(maf, required_cols$i)
+  
+  # return final output table
+  maf = as.data.frame(maf)
+  maf
```

Comment:
`return(maf)` ?

Reply:
I personally don't like this `return` at the end of the function. I restrict its use when the function needs to be interrupted. Is there a recommendation to use it?

Reply:
Curious why would you need to interrupt the function?
On the use of return, this article talks about Python but same logic applies in R: https://www.codecademy.com/forum_questions/558a9ce676b8fe2a77000309

Reply:
For example, Interrupt it when there is an if-else structure.

Reply:
Can you call stop() in this case?

Reply:
`stop` would print an error message.

> For example, Interrupt it when there is an if-else structure
Example:
```
function(...){
  ...
  if(test1){
    return(a)
  }
  b
}
```
Maybe I should replace the word interrupt to terminate.

Reply:
I don't see the point in https://www.codecademy.com/forum_questions/558a9ce676b8fe2a77000309 here.

Reply:
I just would like to say that adding `return` to the end of function is not a problem for me. It's just a personal preference to not use it. But for me there is no difference using it or not.

Reply:
Sorry, maybe I missed what is the goal here and derailed whole conversation 😅
The example you posted here 
function(...){
  ...
  if(test1){
    return(a)
  }
  b
}
will return a and not execute the else part or any code after return statement. 
placing return() somewhere in the function (the if in this example or the very end) will allow storing the generated output in a variable and to be reused downstream.

Reply:
not using return is the same.
```
> rm(list=ls())
> f <- function() 1
> a <- f()
> a
[1] 1
```

Reply:
just out of curiosity:
```
> rm(list=ls())
> g <- function() print(1)
> b <- g()
[1] 1
> b
[1] 1
```